internal/k8s/webhooks/webhookv1beta1/ipaddresspool_webhook.go

```diff
@@ -115,6 +120,12 @@ func validateIPAddressPoolCreate(ipAddress *v1beta1.IPAddressPool) error {
 		level.Error(Logger).Log("webhook", "ipAddress", "action", "create", "name", ipAddress.Name, "namespace", ipAddress.Namespace, "error", err)
 		return err
 	}
+
+	err = validatePoolPerBGPAdv(ipAddress)
```

Comment:
any validation should go directly into validate.Validate. In general, the webhook logic recycles the configuration validation logic that we have in place when we load all the objects.

In this particular case, I suspect collecting the existing bgpadvertisements and passing them should do the trick, as the other path (creating ipaddresspool first, bgpads later) works.

Reply:
I have made the changes.